veps/sig-network/ordinal-naming-scheme-upgrade-path.md

```diff
@@ -0,0 +1,190 @@
+It will then convert any ordinal names found in the target fields to hashed names.
+
+#### Example
+Input:
+
+```xml
+<!-- Interface connected to pod network -->
+<interface type='ethernet'>
+  <mac address='02:51:6f:30:80:10'/>
+  <target dev='tap0' managed='no'/>
+  <model type='virtio-non-transitional'/>
+  <mtu size='1430'/>
+  <alias name='ua-default'/>
+  <rom enabled='no'/>
+  <address type='pci' domain='0x0000' bus='0x01' slot='0x00' function='0x0'/>
+</interface>
+<!-- Interface connected to secondary network -->
+<interface type='ethernet'>
+  <mac address='02:51:6f:30:80:11'/>
+  <target dev='tap1' managed='no'/>
+  <model type='virtio-non-transitional'/>
+  <mtu size='1400'/>
+  <alias name='ua-sec'/>
+  <rom enabled='no'/>
+  <address type='pci' domain='0x0000' bus='0x02' slot='0x00' function='0x0'/>
+</interface>
+```
+
+Output:
+```xml
+<!-- Interface connected to pod network -->
+<interface type='ethernet'>
+  <mac address='02:51:6f:30:80:10'/>
+  <target dev='tap0' managed='no'/>
+  <model type='virtio-non-transitional'/>
+  <mtu size='1430'/>
+  <alias name='ua-default'/>
+  <rom enabled='no'/>
+  <address type='pci' domain='0x0000' bus='0x01' slot='0x00' function='0x0'/>
+</interface>
+<!-- Interface connected to secondary network -->
+<interface type='ethernet'>
+  <mac address='02:51:6f:30:80:11'/>
+  <target dev='tapadd93534eeb' managed='no'/>
+  <model type='virtio-non-transitional'/>
+  <mtu size='1400'/>
+  <alias name='ua-sec'/>
+  <rom enabled='no'/>
+  <address type='pci' domain='0x0000' bus='0x02' slot='0x00' function='0x0'/>
+</interface>
+```
+
+### VMI Interface Status Update
+
+Following a successful migration, the VMI controller will reconcile the new virt-launcher pod.
+As part of this process, it will update the `VMI.Status.Interfaces[].PodInterfaceName` fields to match the pod's
+actual (and now hashed) interface names.
+
+This status update completes the migration to the modern naming scheme and unblocks NIC hot-unplug functionality for the
+VM.
+
+## API Examples
+
+N/A
+
+## Alternatives
+
+<!--
+Outline any alternative designs that have been considered)
+-->
+
+## Scalability
+
+This proposal does not affect scalability.
+
+## Update/Rollback Compatibility
+
+This logic is designed to allow upgrade for VMs defined prior to v1.0 and should not affect newer VMs.
+
+## Functional Testing Approach
+
+The network domain mutator will undergo unit testing.
+However, validating this upgrade path via e2e testing presents significant challenges.
+It requires provisioning a VM in KubeVirt v0.59 and sequentially performing upgrades for every subsequent version.
+Furthermore, because KubeVirt releases are tightly coupled with specific Kubernetes versions, 
+the underlying cluster would need to be upgraded in tandem.
+Given this complexity and resource overhead, running these tests in upstream CI is not feasible.
```

Comment:
I need clarifications about what we actually expect to cover in the tests.
Do we expect to create a VM in kubviert <v1.0.0, and see different names after upgrading?
And where do we expect to see the differences - in the guest? In the VMI? In the pod?
Thank you

Reply:
@yossisegev, A VM should be created in KubeVirt version < 1.0 connected to one or more secondary networks (non-SR-IOV).
This means the pod will be created with an annotation similar to:
```
k8s.v1.cni.cncf.io/network-status: [
  {"interface":"net1", "name":"test1", "namespace":"default"},
  {"interface":"net2", "name":"test1", "namespace":"other-namespace"}
  ]
```

And the pod will have the eth0, net1 and net2 interfaces.
The system will be upgraded until KubeVirt v1.8 (including K8s versions).
After the upgrade, the `VMI.Status.Interfaces[].PodInterfaceName ` field should have a hashed value and NIC hot-unplug should be possible.

The guest will not be affected by this change.

Reply:
Thanks for the response @orelmisan .
So the requires tests are regression - we need to see that the tests are passing pre/post-v1.0 .
Running all tier-2 tests post-upgrade should give this coverage, as at least one of the tests is verifying removal of a secondary interface that was created on VM setup (and not hot-plugged).
Please let me know if I am missing anything.